cmd/juju/application/deploy.go

```diff
@@ -794,9 +794,41 @@ func (c *DeployCommand) maybeReadLocalBundle() (deployFn, error) {
 	}, nil
 }
 
-func (c *DeployCommand) maybeReadLocalCharm() (deployFn, error) {
+func (c *DeployCommand) maybeReadLocalCharm(apiRoot DeployAPI) (deployFn, error) {
+	//NOTE: Here we select the series using the algorithm defined by the
+	//series selector. This serves to override the algorithm found in
+	//`charmrepo.NewCharmAtPath` which is outdated (but must still be called
+	//since the code is coupled with path interpretation logic which needs
+	//to be factored out).
+
+	ch, err := charm.ReadCharm(c.CharmOrBundle)
+	var series string
+	if err != nil {
```

Comment:
It seems that we are saying "if you can't read the charm or bundle, use the series specified by the CLI"

But if we can't read the charm or bundle, is it worthwhile continuing at all?